src/mesh/voxel_cloud.rs

```diff
@@ -1200,13 +1262,98 @@ impl ScalarField {
                     }
                 }
             }
-
-            // Process the current voxel. If it is outside the volumes, set its
-            // value to be positive, if it's inside, set it to negative.
-            self.voxels[one_dimensional] = if discovered_as_outer_and_empty[one_dimensional] {
-                Some(distance)
-            } else {
-                Some(-distance)
+            let is_outside = discovered_as_outer_and_empty[one_dimensional];
+
+            // Process the current voxel.
+
+            self.voxels[one_dimensional] = match falloff_function {
+                FalloffFunction::Linear(mul) => {
+                    // distance values: 0 to infinity
+                    // suggested multiplier: 1
+                    // output values: 0 to infinity
+                    // value on volume: 0
+                    if distance == 0.0 {
+                        Some(0.0)
+                    } else {
+                        let value = distance * mul;
+                        if is_outside {
+                            Some(value)
+                        } else {
+                            Some(-value)
+                        }
+                    }
+                }
+                FalloffFunction::Square(mul) => {
+                    // distance values: 0 to infinity
+                    // suggested multiplier: 1
+                    // output values: 0 to infinity
+                    // value on volume: 0
+                    if distance == 0.0 {
+                        Some(0.0)
+                    } else {
+                        let value = distance * mul * distance * mul;
```

Comment:
Just a philosophical question, but should here and in `InverseSquare` be `mul` squared also?

Reply:
dunno. i think it's easier to imagine the distance multiplied and then something magic happens with the new value

Reply:
So which do you prefer? Because for polynomials the parameter usually isn't exponentiated, but here it is.

Reply:
I prefer the way it is